openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx

```diff
@@ -82,8 +98,43 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
     return metricsOptions;
   }, [columns]);
 
+  const { partitionColumnOptions, isPartitionDisabled } = useMemo(() => {
+    const partitionColumnOptions = columns.reduce((result, column) => {
+      if (SUPPORTED_PARTITION_TYPE.includes(column.dataType)) {
+        return [
+          ...result,
+          {
+            value: column.name,
+            label: column.name,
+          },
+        ];
+      }
+
+      return result;
+    }, [] as { value: string; label: string }[]);
+    const isPartitionDisabled = partitionColumnOptions.length === 0;
+
+    return {
+      partitionColumnOptions,
+      isPartitionDisabled,
+    };
+  }, [columns]);
+
+  const intervalTypeOptions = useMemo(() => {
+    return Object.values(PartitionIntervalType).map((value) => ({
+      value,
+      label: value,
+    }));
+  }, []);
+  const intervalUnitOptions = useMemo(() => {
+    return Object.values(PartitionIntervalUnit).map((value) => ({
+      value,
+      label: value,
+    }));
+  }, []);
```

Comment:
we can have them in utils or constant files

Reply:
updated!